dev-support/Jenkinsfile_GitHub

```diff
@@ -493,22 +495,19 @@ pipeline {
                         SKIP_ERRORPRONE = true
                     }
                     when {
-			allOf {
-                            // this will return true if the pipeline is building a change request, such as a GitHub pull request.
-                            changeRequest()
-                            expression { env.CHANGE_TARGET in ['master', 'branch-3'] }
```

Comment:
Should be `when in [master, branch-3, branch-2]` ?

Reply:
Will change this part later in HBASE-28678. We still have some problem for jdk17 builds, for example, we do not unstash the result later so on the final jira report we missed the jdk 17 line...